File: server/read/fastify/src/models/CommentReadData.ts
import { Comment } from '../entities/Comment';

export class CommentReadData {
  articleUUID: string;
  uuid: string;
  createdByUUID: string;
  content: string;
  createdAt: Date;
  updatedAt: Date;

  constructor(private comment: Comment) {
    this.uuid = comment.uuid;
    this.content = comment.content;
    this.createdAt = comment.createdAt;
    this.updatedAt = comment.updatedAt;
  }

  async lazyFill() {
    const article = await this.comment.article;
    this.articleUUID = article.uuid;
    const createdBy = await this.comment.createdBy;
    this.createdByUUID = createdBy.uuid;
  }
}

File: server/write/spring/src/main/kotlin/com/board/server/write/spring/models/CommentReadData.kt
package com.board.server.write.spring.models

import com.board.server.write.spring.entities.CommentEntity
import java.time.LocalDateTime
import java.util.*

data class CommentReadData(
    private val comment: CommentEntity,
    val articleUUID: UUID,
    val createdByUUID: UUID,
) {
    val uuid: UUID = comment.uuid
    val content: String = comment.content
    val createdAt: LocalDateTime = comment.createdAt
    val updatedAt: LocalDateTime = comment.updatedAt
}
